js/ui/iconGrid.js

```diff
@@ -569,6 +569,19 @@ const IconGrid = new Lang.Class({
                 this.actor.set_skip_paint(children[i], false);
             }
         }
+
+        /* When we're running on small screens, to make it fit 5 columns
+         * on the available space, we shall reduce the icon size. Do
+         * this by adding (or removing, case the screen is big enought)
+         * the .composite-mode style class. Given that the height of the
+         * widget varies according to the number of icons, we only check
+         * the width here.
+         */
+         if (availWidth < 800) {
```

Comment:
Tracking the allocation here, while probably effective, sounds a bit fragile. Do you think you could instead track the size of the primary monitor as defined in `layout.js`?